fastpy/fastpy/run/Runner.py

```diff
@@ -0,0 +1,99 @@
+import os
+import shutil
+import itertools
+from subprocess import call
+
+from fastpy.io.config import load_json_config, store_json_config
+from fastpy.utils import get_date_time_tag
+from common import DATA_DIR_PATH, PROJECT_ROOT_PATH
+
+# This mapping will be used to pass command line arguments to benchmark executable
+PARAM_TO_C_MAP = {'algorithm':  '-a',
+                  'obj_func':   '-o',
+                  'dimension':  '-d',
+                  'n_iter':     '-n',
+                  'population': '-p',
+                  'min_val':    '-y',
+                  'max_val':    '-z'}
+
+BENCHMARK_BIN_DIR = os.path.join(PROJECT_ROOT_PATH, '../benchmark')
+BENCHMARK_BIN = 'benchmark'
+
+TIMING_OUT_FILE = 'timings.csv'
+SOLUTION_OUT_FILE = 'solution.csv'
+
+
+class BenchmarkRunner:
+
+    def __init__(self, config_path, data_dir=DATA_DIR_PATH):
+        self.config = load_json_config(config_path)
+        self.data_dir = data_dir
+        self.timestamp = get_date_time_tag()
+
+    def run_benchmarks(self):
+        """Main wrapper function to loop over all possible parameter combinations."""
+
+        os.mkdir(self._output_dir)  # main output dir for this whole run
+        shutil.copy(os.path.join(BENCHMARK_BIN_DIR, BENCHMARK_BIN), self._output_dir)
+        store_json_config(self.config, self._output_dir, 'config.json')
+
+        for run_idx, run_config in enumerate(self._build_param_sets()):
+
+            sub_dir = os.path.join(self._output_dir, f'run_{run_idx}')  # sub output dir for one param combo
+            os.mkdir(sub_dir)
+
+            store_json_config(run_config, sub_dir, 'run_config.json')
+
+            self._run_algorithm(run_config, sub_dir)
+
+    def _run_algorithm(self, run_config, sub_dir):
+        """Subprocess call to run a single algorithm."""
+
+        call_str = ' '.join([self._benchmark_bin, self._create_params_str(run_config)])
+        call_str += ' -f ' + os.path.join(sub_dir, TIMING_OUT_FILE)
+        call_str += ' -s ' + os.path.join(sub_dir, SOLUTION_OUT_FILE)
+
+        call(call_str, shell=True)
```

Comment:
Maybe use `check_call` to ensure no error occurred in the C executable? I feel like this could very well happen due to seg faults etc that we miss during unit testing etc.

Reply:
Good one, will do.